nomad/structs/structs.go

```diff
@@ -3444,7 +3462,12 @@ func validateServices(t *Task) error {
 		knownServices[service.Name+service.PortLabel] = struct{}{}
 
 		if service.PortLabel != "" {
-			servicePorts[service.PortLabel] = append(servicePorts[service.PortLabel], service.Name)
+			if _, err := strconv.Atoi(service.PortLabel); service.AddressMode == "driver" && err == nil {
```

Comment:
This is hard to read, and I am not sure if the logic is correct. If addressmode is not driver, and the strconv returns an error (if portlabel is *not* an integer), then should the error really be ignored? if so, why even do the parsing?